__tests__/Record.ts

```diff
@@ -253,4 +253,25 @@ describe('Record', () => {
       ['b', 20],
     ]);
   });
+
+  it('check that reset does reset the record. See https://github.com/immutable-js-oss/immutable-js/issues/85 ', () => {
+    type UserType = {
+      name: string;
+      roles: List<string> | Array<string>;
+    };
+
+    const User = Record<UserType>({
+      name: 'default name',
+      roles: List<string>(),
+    });
+
+    const user_0 = new User({
+      name: 'John',
+      roles: ['superuser', 'admin'],
+    });
+    const user_1 = user_0.clear();
+
+    expect(user_1.name).toBe('default name');
+    expect(user_1.roles).toEqual(List());
```

Comment:
The problem occurred when you add:
```
const user_3 = user_0.withMutations(mutable => mutable.clear());
expect(user_3.roles).toEqual(List());
```

Can you try adding that and see if it still works?

Reply:
In the default issue, `roles` is an array, not an immutable List object but yes I will add this test case to be sure that it works as expected

Reply:
You are right : calling clear on is mutable does not works anymore. I will investigate on this